lib/galaxy/tools/__init__.py

```diff
@@ -1721,6 +1724,16 @@ def get_static_param_values(self, trans):
                 args[key] = model.User.expand_user_properties(trans.user, param.value)
             else:
                 raise Exception("Unexpected parameter type")
+        if self.tool_type == 'data_source_async':
+            # parse xml in order to get 'environment_variable' values
+            for user_env_var in XmlToolSource(raw_tool_xml_tree(self.config_file)).parse_environment_variables():
```

Comment:
Re-parsing the XML is not great, ``self.tool.environment_variables`` would be a lot better here.

Reply:
ok, thank you for pointing this to me, I was not aware of this. I'm trying with a galaxy 20_05. Is this `self.tool.environment_variables` available? Because when I try to print its values, I've got an error `AttributeError: tool`
I will convert this to a draft, and then I will try to find time to work again on that, but based on 21_01!

Reply:
I guess you're in the tool so it would just be ``self.environment_variables`` - sorry about that.